src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/ProteinsList.tsx

```diff
@@ -14,50 +14,38 @@
  * limitations under the License.
  */
 
-import React, { useEffect, useState } from 'react';
+import React from 'react';
 
 import ProteinsListItem from './proteins-list-item/ProteinsListItem';
 
-import { fetchGene } from 'src/content/app/entity-viewer/shared/rest/rest-data-fetchers/geneData';
-import { getLongestProteinLength } from 'src/content/app/entity-viewer/shared/helpers/entity-helpers';
+import {
+  getLongestProteinLength,
+  isProteinCodingTranscript
+} from 'src/content/app/entity-viewer/shared/helpers/entity-helpers';
 import { defaultSort } from 'src/content/app/entity-viewer/shared/helpers/transcripts-sorter';
 
 import { Gene } from 'src/content/app/entity-viewer/types/gene';
 
 import styles from './ProteinsList.scss';
 
 type ProteinsListProps = {
-  geneId: string;
+  gene: Gene;
 };
 
 type ProteinsListWithDataProps = {
   gene: Gene;
 };
 
 const ProteinsList = (props: ProteinsListProps) => {
-  const [geneData, setGeneData] = useState<Gene | null>(null);
-
-  useEffect(() => {
-    const abortController = new AbortController();
-
-    fetchGene(props.geneId, abortController.signal).then((result) => {
-      if (result) {
-        setGeneData(result);
-      }
-    });
-
-    return function cleanup() {
-      abortController.abort();
-    };
-  }, [props.geneId]);
-
-  return geneData ? <ProteinsListWithData gene={geneData} /> : null;
+  // TODO: either consider making the graphql request directly from here,
```

Comment:
Even if we decide to make the graphql request here, can we still not combine these two components into one and make use the `loading` state from useQuery?

Reply:
Merged them into a single component.